From ecb039305de15bb0baa1ac6a78980cb2aca992fb Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Tue, 7 Feb 2023 12:13:18 +0100
Subject: [PATCH 0337/1695] drivers: stm32_exti: stm32mp_exti: drop compile
 macro CFG_EMBED_DTB

All the targets that use this driver define CFG_EMBED_DTB and use
the devicetree.

Drop the check for CFG_EMBED_DTB defined as it is always true.

While there, update the date in the copyright text.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f010cae17ed40fe2bfdac1d7401c9f7d1d82f1d2 drivers: stm32_exti: stm32mp_exti: drop compile macro CFG_EMBED_DTB]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I2c78159fb509b46b34e3476a3c6187c1819a6caf
---
 core/drivers/stm32_exti.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/core/drivers/stm32_exti.c b/core/drivers/stm32_exti.c
index b3f701734..ff41bf6d7 100644
--- a/core/drivers/stm32_exti.c
+++ b/core/drivers/stm32_exti.c
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: BSD-3-Clause
 /*
- * Copyright (c) 2021, STMicroelectronics
+ * Copyright (c) 2021-2023, STMicroelectronics
  */
 
 #include <drivers/stm32_exti.h>
@@ -343,7 +343,6 @@ stm32_exti_pm(enum pm_op op, unsigned int pm_hint __unused,
 	return TEE_SUCCESS;
 }
 
-#ifdef CFG_EMBED_DTB
 static TEE_Result stm32_exti_probe(const void *fdt, int node,
 				   const void *comp_data __unused)
 {
@@ -373,4 +372,3 @@ DEFINE_DT_DRIVER(stm32_exti_dt_driver) = {
 	.match_table = stm32_exti_match_table,
 	.probe = &stm32_exti_probe,
 };
-#endif
-- 
2.47.2

